auth.py

```diff
@@ -59,11 +59,31 @@ def decorated_function(**kwargs):
             if access is None:
                 return redirect(url_for("auth.login"))
             else:
-                if Config.ROLE.get(access) is not None:
-                    if Config.ROLE.get(access)[0] <= Config.ROLE.get(role)[0] and access != role:
-                        return redirect(Config.ROLE.get(access)[1])
-                else:
+                if role == 'user':
+                    userevent_id = kwargs.get('id')
+                    if 'user_info' in session:
+                        if 'uiucedu_is_member_of' in session.get('user_info'):
+                            for member in session.get('user_info').get('uiucedu_is_member_of'):
+                                items = member.split( )
+                                if len(items) == 5:
+                                    if items[2] == 'user' and items[4] == 'admins':
+                                        if userevent_id is None:
+                                            return view(**kwargs)
+                                        else:
+                                            event = find_one(current_app.config['EVENT_COLLECTION'],
+                                                             condition={"_id": ObjectId(userevent_id)},
+                                                             projection={'createdByGroupId': 1})
+                                            if 'createdByGroupId' in event:
+                                                for admin_group in session['groups']:
```

Comment:
@AnubhavG1999 I need your pr to provide a list of group information inside the session.